src/main/java/org/opensearch/security/resolver/IndexResolverReplacer.java

```diff
@@ -750,11 +752,16 @@ private boolean getOrReplaceAllIndices(final Object request, final IndicesProvid
             }
             ((IndexRequest) request).index(newIndices.length != 1 ? null : newIndices[0]);
         } else if (request instanceof Replaceable) {
-            String[] newIndices = provider.provide(((Replaceable) request).indices(), request, true);
-            if (checkIndices(request, newIndices, false, allowEmptyIndices) == false) {
+            String[] newIndicesOrAliases;
+            if (request instanceof GetAliasesRequest) {
```

Comment:
I see that [GetAliasesRequest implements AliasesRequest](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/admin/indices/alias/get/GetAliasesRequest.java#L49). Can this use the more generic AliasesRequest? What other AliasesRequests are there and would they be impacted by the same issue?

Reply:
Looks like the only other AliasesRequest I could find is within the IndicesAliasRequest: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/admin/indices/alias/IndicesAliasesRequest.java#L100. From the description it is: `A request to add/remove aliases for one or more indices.`. IMO we can change this to `instaceof AliasesRequest`, wdyt?

Reply:
`AliasesRequests` extends `IndicesRequest.Replaceable` which means an `AliasesRequest` is also an `IndicesRequest`. In that case would the `provider.provide` call need to consider both `.aliases()` and `.indices()` from the request object?

Is it possible that a GetAliasesRequest can contain `.aliases()` and `.indices()` at the same time?

Reply:
+1 . Will there be a case where indices() and aliases() are present at the same time. Which should be given higher priority in that case or should they both be considered ?